include/openthread/platform/radio.h

```diff
@@ -694,6 +694,46 @@ void otPlatRadioClearSrcMatchShortEntries(otInstance *aInstance);
  */
 void otPlatRadioClearSrcMatchExtEntries(otInstance *aInstance);
 
+/**
+ * Get the radio supported channel mask that the device is allowed to be on.
+ *
+ * @param[in]  aInstance   The OpenThread instance structure.
+ *
+ * @returns The radio supported channel mask.
+ *
+ */
+uint32_t otPlatRadioGetSupportedChannelMask(otInstance *aInstance);
+
+/**
+ * Get the radio preferred channel mask that the device prefers to form on.
+ *
+ * @param[in]  aInstance   The OpenThread instance strucyyture.
+ *
+ * @returns The radio preferred channel mask.
+ *
+ */
+uint32_t otPlatRadioGetPreferredChannelMask(otInstance *aInstance);
+
+/**
+ * Get the minimum channel number of IEEE 802.15.4 physical layer.
+ *
+ * @param[in]  aInstance   The OpenThread instance strucyyture.
+ *
+ * @returns The minimum channel number.
+ *
+ */
+uint8_t otPlatRadioGetPhyChannelMin(otInstance *aInstance);
```

Comment:
I think we should submit a separate PR to remove these two functions. Just use the channel mask for all cases. I think we should reduce APIs.

Reply:
Agree, deleted the two APIs.